controllers/datadogagent/controller_reconcile_v2_common.go

```diff
@@ -92,12 +92,12 @@ func (r *Reconciler) createOrUpdateDeployment(parentLogger logr.Logger, dda *dat
 		now := metav1.NewTime(time.Now())
 		err = kubernetes.UpdateFromObject(context.TODO(), r.client, updateDeployment, currentDeployment.ObjectMeta)
 		if err != nil {
-			updateStatusFunc(nil, newStatus, now, metav1.ConditionFalse, "CreateFailed", "Unable to Update Deployment")
+			updateStatusFunc(nil, newStatus, now, metav1.ConditionFalse, "UpdateFailed", "Unable to update Deployment")
```

Comment:
I think in that case the condition should be `True` 
```suggestion
			updateStatusFunc(nil, newStatus, now, metav1.ConditionTrue, "UpdateFailed", "Unable to update Deployment")
```

Reply:
Also in the `else` you should set the opposite